### DISCUSSION TITLE
Elastalert+Jira CA cert (CERTIFICATE_VERIFY_FAILED)

### DISCUSSION THREAD

**Discussion by alksegrv** [jertel/elastalert2#1058] [Q&A] [answered]
I am using a self-signed certificate when connecting to Jira. How to remove the "CERTIFICATE_VERIFY_FAILED" 

my rule
`name: test
type: frequency
index: logstash-dns-*
num_events: 2
timeframe:
  minutes: 5
filter:
  - term:
      dns.rrname: "yotube.ru"
alert:
  - "jira"
jira_server: "https://jt.kryptonite.ru:443"
jira_project: "EA"
jira_issuetype: "Task"
jira_account_file: "/opt/elastalert/jiracredentials.yaml"`

error text:

WARNING:jira.resilientsession:Got ConnectionError [HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url:                          /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable                          to get local issuer certificate (_ssl.c:992)')))] errno:None on GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo

WARNING:jira.resilientsession:Got recoverable error from GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo, will retry [1/3] in 15.61                         1105847577193s. Err: HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused                          by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certifica                         te (_ssl.c:992)')))
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:39:                         16 UTC)" skipped: maximum number of running instances reached (1)
WARNING:jira.resilientsession:Got ConnectionError [HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))] errno:None on GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo

WARNING:jira.resilientsession:Got recoverable error from GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo, will retry [2/3] in 24.916440417987204s. Err: HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:39:26 UTC)" skipped: maximum number of running instances reached (1)
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:39:36 UTC)" skipped: maximum number of running instances reached (1)
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:39:46 UTC)" skipped: maximum number of running instances reached (1)
WARNING:jira.resilientsession:Got ConnectionError [HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))] errno:None on GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo

WARNING:jira.resilientsession:Got recoverable error from GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo, will retry [3/3] in 33.40210851140461s. Err: HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:39:56 UTC)" skipped: maximum number of running instances reached (1)
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:40:06 UTC)" skipped: maximum number of running instances reached (1)
WARNING:apscheduler.scheduler:Execution of job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:40:16 UTC)" skipped: maximum number of running instances reached (1)
ERROR:apscheduler.executors.default:Job "Internal: Handle Config Change (trigger: interval[0:00:10], next run at: 2022-12-21 13:40:26 UTC)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 1042, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 414, in connect
    self.sock = ssl_wrap_socket(
                ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ssl.py", line 517, in wrap_socket
    return self.sslsocket_class._create(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ssl.py", line 1075, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.11/ssl.py", line 1346, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/elastalert.py", line 1196, in handle_config_change
    self.load_rule_changes()
  File "/usr/local/lib/python3.11/site-packages/elastalert/elastalert.py", line 1078, in load_rule_changes
    new_rule = self.rules_loader.load_configuration(rule_file, self.conf)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/loaders.py", line 229, in load_configuration
    self.load_modules(rule, args)
  File "/usr/local/lib/python3.11/site-packages/elastalert/loaders.py", line 493, in load_modules
    rule['alert'] = self.load_alerts(rule, alert_field=rule['alert'])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/loaders.py", line 527, in load_alerts
    alert_field = [create_alert(a, b) for a, b in alert_field]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/loaders.py", line 527, in <listcomp>
    alert_field = [create_alert(a, b) for a, b in alert_field]
                   ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/loaders.py", line 518, in create_alert
    return alert_class(alert_config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/elastalert/alerters/jira.py", line 111, in __init__
    self.client = JIRA(self.server, basic_auth=(self.user, self.password))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/jira/client.py", line 574, in __init__
    si = self.server_info()
         ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/jira/client.py", line 3059, in server_info
    j = self._get_json("serverInfo")
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/jira/client.py", line 3764, in _get_json
    r = self._session.get(url, params=params)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/jira/resilientsession.py", line 222, in request
    raise exception
  File "/usr/local/lib/python3.11/site-packages/jira/resilientsession.py", line 202, in request
    response = super().request(method, url, **processed_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 563, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))
WARNING:jira.resilientsession:Got ConnectionError [HTTPSConnectionPool(host='jt.kryptonite.ru', port=443): Max retries exceeded with url: /rest/api/2/serverInfo (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)')))] errno:None on GET https://jt.kryptonite.ru:443/rest/api/2/serverInfo

**Comment by nsano-rururu** [comment, marked as answer]
https://github.com/Yelp/elastalert/issues/451
https://stackoverflow.com/questions/23961956/ssl3-certificate-verify-failed-when-connecting-to-jira-api-using-python
